### ISSUE TITLE
[Bug]: Redundant segments are restored, after querynode down

### ISSUE THREAD

**Issue by xige-16** [milvus-io/milvus#12340]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster):
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

querycoord will recover all the segment of the partition the down node has loaded

### Expected Behavior

querycoord only recover the segment the down node has loaded

**Comment by xiaofan-luan** [comment]
what is the difference between current behaviour and Expected Behavior

**Comment by xige-16** [comment]
The issue should has been fixed by pr #13543

**Comment by xige-16** [comment]
verified
